### PR TITLE
Add support for doctrine/persistence 2.x

### PR DESCRIPTION
**Pull request by awurth**
## Subject

Allow version 2 for "doctrine/persistence".

I am targeting this branch, because this change respects BC.

## Changelog

```markdown
### Added
- Added support for "doctrine/persistence:^2.0".
```

### CODE REVIEW

**Comment by jordisala1991** [comment]
It would be nice to have doctrine / persistence as a dev dependency at least (supporting both 1.3 and 2)

File: src/Entity/CategoryManager.php [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 namespace Sonata\ClassificationBundle\Entity;
 
-use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\Persistence\ManagerRegistry;
```
Comment (core23) [inline src/Entity/CategoryManager.php:16]: Changing this, requires `doctrine/perstience` as a non-dev dependency

**Review by core23 — CHANGES_REQUESTED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**